atomicfu/src/jvmMain/kotlin/kotlinx/atomicfu/parking/KThread.kt

```diff
@@ -0,0 +1,19 @@
+package kotlinx.atomicfu.parking
+
+actual class KThread internal actual constructor() {
+    internal val parker = ThreadParker(JvmParkingDelegator())
+    actual companion object {
+        actual fun currentThread(): KThread = localKThread.get()
+    }
+}
+
+actual class Parker actual private constructor() {
+    actual companion object {
+        actual fun park() = localKThread.get().parker.park()
+        actual fun parkNanos(nanos: Long) = localKThread.get().parker.parkNanos(nanos)
+        actual fun unpark(kThread: KThread) = localKThread.get().parker.unpark()
+    }
+}
+
+private val localKThread = ThreadLocal<KThread>().also { it.set(KThread()) }
```

Comment:
This won't work, you need `ThreadLocal<KThread>.withInitialValue { KThread() }`. The reason: `.also { it.set(KThread()) }` will only be called once, on the thread that is the first to access `localKThread`, whereas the other threads will all have the default value, which is `null`.